packages/nimble-components/src/select/index.ts

```diff
@@ -49,51 +146,701 @@ export class Select extends FoundationSelect implements ErrorPattern {
+        super.keydownHandler(e);
+        const key = e.key;
+        if (e.ctrlKey || e.shiftKey) {
+            return true;
+        }
+
+        switch (key) {
+            case keySpace: {
+                // when dropdown is open allow user to enter a space for filter text
+                if (this.open && this.filterMode !== FilterMode.none) {
+                    break;
+                }
+
+                e.preventDefault();
+                if (this.collapsible && this.typeAheadExpired) {
+                    this.open = !this.open;
+                    this.focusFilterInput();
+                }
+                break;
+            }
+            case keyHome:
+            case keyEnd: {
+                e.preventDefault();
+                break;
+            }
+            case keyEnter: {
+                e.preventDefault();
+                if (
+                    this.filteredOptions.length === 0
+                    || this.filteredOptions.every(o => o.disabled)
+                ) {
+                    return false;
+                }
+                this.updateSelectedIndexFromFilteredSet();
+                this.open = !this.open;
+                this.focusFilterInput();
+                break;
+            }
+            case keyEscape: {
+                // clear filter as update to "selectedIndex" will result in processing
+                // "options" and not "_options"
+                this.filter = '';
+                if (this.committedSelectedOption) {
+                    this.clearSelection();
+                    this.selectedIndex = this._options.indexOf(
+                        this.committedSelectedOption
+                    );
+                }
+                if (this.collapsible && this.open) {
+                    e.preventDefault();
+                    this.open = false;
+                }
+                // reset 'selected' state otherwise the selected state doesn't stick.
+                const selectedOption = this._options[this.selectedIndex];
+                if (selectedOption) {
+                    selectedOption.selected = true;
+                }
+                break;
+            }
+            case keyTab: {
+                if (this.collapsible && this.open) {
+                    e.preventDefault();
+                    this.open = false;
+                }
+
+                return true;
+            }
+
+            default: {
+                break;
+            }
+        }
+
+        if (!this.open && this.indexWhenOpened !== this.selectedIndex) {
+            this.updateValue(true);
+            this.indexWhenOpened = this.selectedIndex;
+        }
+
+        return !(key === keyArrowDown || key === keyArrowUp);
+    }
+
+    /**
+     * Updates the proxy value when the selected index changes.
+     *
+     * @param prev - the previous selected index
+     * @param next - the next selected index
+     *
+     * @internal
+     */
+    public override selectedIndexChanged(
```

Comment:
@jattasNI @rajsite, everything from here until `openChanged` _except_ for `setSelectedOptions` is copied from FAST.